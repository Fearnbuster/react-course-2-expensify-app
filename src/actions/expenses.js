
import firestore from '../firebase/firebase';

export const addExpense = (expense) => ({
  type: 'ADD_EXPENSE',
  expense
});

export const startAddExpense = (expenseData = {}) => {
  return (dispatch) => {
    const {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
    } = expenseData;

    const expense = { description, note, amount, createdAt };

    return firestore.collection('expenses')
      .add(expense)
      .then((newExpense) => {
        dispatch(addExpense({
          id: newExpense.id,
          ...expense
        }));
      });
  };
};

export const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id,
  updates
});

export const removeExpense = ({ id }) => ({
  type: 'REMOVE_EXPENSE',
  id
});

