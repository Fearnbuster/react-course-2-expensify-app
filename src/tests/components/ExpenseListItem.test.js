
import React from 'react';
import { shallow } from 'enzyme';

import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

describe('componenets/ExpenseListItem', () => {
  it('should render ExpenseListItem with provided values', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)

    expect(wrapper).toMatchSnapshot();
  });
});