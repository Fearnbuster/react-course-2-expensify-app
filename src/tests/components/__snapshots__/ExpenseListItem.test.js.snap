// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componenets/ExpenseListItem should render ExpenseListItem with provided values 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/1"
    >
      Gum
    </Link>
  </h3>
  <p>
    Amount: 
    195
  </p>
  <p>
    Date: 
    0
  </p>
</div>
`;
